### DISCUSSION TITLE
Authorization via telegram

### DISCUSSION THREAD

**Discussion by JekaSylar** [westacks/telebot#72] [Q&A]
Hello, sorry for one more question, I don’t understand how to log in via telegram
````php
// Авторизация через телеграмм бота

Route::post('/auth/telegram/action', [AuthController::class, 'authTelegram'])->middleware(
    'telebot-webapp'
)->name('login.telegram');

  public function authTelegram(Request $request)
    {
        $user_id = $request->telegramWebAppUser('id');

        dd($user_id);
    }
```
Gives an error message Target class [telebot-webapp] does not exist

**Comment by punyflash** [comment]
`telebot-webapp` is a middleware to get Telegram users from [Telegram Web Apps](https://core.telegram.org/bots/webapps). If you want to login users to your application using Telegram, you need to use [Telegram Login Widget](https://core.telegram.org/widgets/login). It has pretty similar auth mechanism based on bot token. However you will need to implement Auth guard by yourself. Or use [Laravel Socialite](https://socialiteproviders.com/Telegram/)
